src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -476,6 +479,26 @@ public static void ValidateReferenceAssemblies(IEnumerable<IPackageFile> files,
             }
         }
 
+        public static void ValidateLicenseFile(IEnumerable<IPackageFile> files, LicenseMetadata licenseMetadata)
+        {
+            if (licenseMetadata?.Type == LicenseType.File)
+            {
+                var ext = Path.GetExtension(licenseMetadata.License);
+                if (!string.IsNullOrEmpty(ext) &&
+                        !ext.Equals(".txt", StringComparison.OrdinalIgnoreCase) &&
+                        !ext.Equals(".md", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new PackagingException(NuGetLogCode.NU5031, string.Format(CultureInfo.CurrentCulture, NuGetResources.Manifest_LicenseFileExtensionIsInvalid, licenseMetadata.License));
+                }
+                var strippedLicenseFileLocation = PathUtility.StripLeadingDirectorySeparators(licenseMetadata.License);
+                var count = files.Where(e => PathUtility.StripLeadingDirectorySeparators(e.Path).Equals(strippedLicenseFileLocation)).Count();
```

Comment:
case sensitivity check here, also normalize path separators before comparing paths.
also, i am not sure what's the point of this check. If the LicenseMetadata in the PackageBuilder had a LicenseFile , it's NuGet's responsibility to make sure it was added to the list of Files in the PackageBuilder. Why would the count here ever be 0?

If the count is more than 1, then i would imagine this is an error.

Reply:
Refer to https://github.com/NuGet/NuGet.Client/pull/2450#discussion_r223155993. 

Will normalize the separators before doing the stripping and comparing paths.